lib/util/flattenOperands.js

```diff
@@ -35,6 +37,14 @@ interested in how that works
 // Input: an expression tree
 // Output: the expression tree updated with flattened operations
 function flattenOperands(node) {
+  // If the node is a mixed number, do not perform any flattening
+  // Flattening will take out the implicit multiplication, and so
+  // it will be impossible to tell if the node is a mixed number or
+  // if it is legitimate multiplication
```

Comment:
good call!

maybe add an extra comment that says that converting fractions happens before any other step, so the tree _will_ get flattened before any other changes happen